---
title: "Scerevisiae RBP broad motif search in Calbicans"
author: "Paddy"
date: "24/02/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(here)
library(tidyverse)
library(cowplot)
theme_set(theme_cowplot(font_size = 12))
library(Biostrings)
```

```{r setwd_for_interactive_sessions_PHmac, include=FALSE,eval=FALSE}
setwd("~/Documents/SENIOR HONOURS/Honours Project 20:21/Motif analysis/FASTA sequences/All genes")
here::here() 
```

```{r load_upstream1000}
# define fasta filename
Ca_up1000_file <- paste(here::here(), "Calbicans_SC5314_ATG_upstream_1000nt.fasta",
                        sep= "/")
# load fasta file as DNA string set
Ca_up1000_DSS <- readDNAStringSet(Ca_up1000_file)
# remove truncated sequences, which we don't need.
Ca_up1000_DSS <- Ca_up1000_DSS[width(Ca_up1000_DSS) == 1000]
# assign just the ORF id to the name, but keep all the info. 
Ca_names_up_info <- 
  tibble(everything = names(Ca_up1000_DSS)) %>%
  tidyr::separate(everything,into = c("id","strain","info","seqtype","seq","length"), sep = " \\| ") %>%
  dplyr::select(id, info,seq)
names(Ca_up1000_DSS) <- Ca_names_up_info$id
# print the DNA string set to check it came out ok
Ca_up1000_DSS
```

```{r load_downstream1000}
# define fasta filename
Ca_down1000_file <- paste(here::here(), "Calbicans_SC5314_stopcodon_downstream_1000nt.fasta",
                        sep= "/")
# load fasta file as DNA string set
Ca_down1000_DSS <- readDNAStringSet(Ca_down1000_file)
# remove truncated sequences, which we don't need.
Ca_down1000_DSS <- Ca_down1000_DSS[width(Ca_down1000_DSS) == 1000]
# assign just the ORF id to the name, but keep all the info. 
Ca_names_down_info <- 
  tibble(everything = names(Ca_down1000_DSS)) %>%
  tidyr::separate(everything,into = c("id","strain","info","seqtype","seq","length"), sep = " \\| ") %>%
  dplyr::select(id, info,seq)
names(Ca_down1000_DSS) <- Ca_names_down_info$id
# print the DNA string set to check it came out ok
Ca_down1000_DSS
```


```{r CNYTCNYT_count_up}
CNYTCNYT_count_up_df <- tibble(id = Ca_names_up_info$id,
                            info = Ca_names_up_info$info,
                            count_up1000 = vcountPattern(pattern = DNAString("CTTGTAG"),
                                 subject = Ca_up1000_DSS,
                                 fixed = "subject"),
                            count_up200 = vcountPattern(pattern = DNAString("CTTGTAG"),
                                 subject = subseq(Ca_up1000_DSS,start = 801L, end = 1000L),
                                 fixed = "subject"),
                            count_up100 = vcountPattern(pattern = DNAString("CTTGTAG"),
                                 subject = subseq(Ca_up1000_DSS,start = 901L, end = 1000L),
                                 fixed = "subject")
)
summary(CNYTCNYT_count_up_df)
```
```{r high_counts_up, results="asis"}
arrange(CNYTCNYT_count_up_df,desc(count_up100)) %>%
    head(n = 20) %>%
    knitr::kable()
```
```{r CNYTCNYT_count_down}
CNYTCNYT_count_down_df <- tibble(id = Ca_names_down_info$id,
                            info = Ca_names_down_info$info,
                            count_down1000 = vcountPattern(pattern = DNAString("CTTGTAG"),
                                 subject = Ca_down1000_DSS,
                                 fixed = "subject"),
                            count_down200 = vcountPattern(pattern = DNAString("CTTGTAG"),
                                 subject = subseq(Ca_down1000_DSS,start = 1L, end = 200L),
                                 fixed = "subject"),
                            count_down100 = vcountPattern(pattern = DNAString("CTTGTAG"),
                                 subject = subseq(Ca_down1000_DSS,start = 1L, end = 100L),
                                 fixed = "subject")
)
summary(CNYTCNYT_count_down_df)
```
```{r high_counts_down, results="asis"}
arrange(CNYTCNYT_count_down_df,desc(count_down100)) %>%
    head(n = 20) %>%
    knitr::kable()
```

